test/integ/ezrassor_topic_switch/test_topic_switch.py

```diff
@@ -55,7 +53,7 @@ def generate_test_description(ready_fn):
                     "channels": TEST_CHANNELS,
                 }.items(),
             ),
-            launch.actions.OpaqueFunction(function=lambda context: ready_fn()),
+            launch_testing.actions.ReadyToTest(),
```

Comment:
How does this work? Was ready_fn a function that pytest would pass in?

Reply:
yes I still am not really sure how it works under the hood, but ROS would pass in a function and we were supposed to execute it like so. But now in Foxy they simplified it to only require `ReadyToTest`. I believe it's to support usecases where you shouldn't launch the nodes immediately and instead do some work first.